### DISCUSSION TITLE
Any ideas on porting an online game from COCOS2D to Stride. (C++)

### DISCUSSION THREAD

**Discussion by xheox** [stride3d/stride#1476] [Q&A] [answered]
Hi, 
I discovered Stride today, 
Very neat sofar ... 
I have fairly big source code for a card game written in COCOS2D (C++). 
I would like to port it to .NET to have a better code (maitainable) and more reliable client server connections.
I have 2 questions : 
- Is there anything to take in account for STRIDE in particular if I am going to re-use some existing C++  code . I am thinking Interop or Creating a wrapper like other type of apps in C# consuming C++ code.
- I did not see anything about multi-player online games in Stride. So, do I have to build the server from scratch ? what would be the recommended way to do it ? I am looking for a simple server API to manage connections, and have a good set of primitives for messaging and event publishing. If there  is nothing, then I might look for message broker or an actor framework (my preferred). 

Thanks

**Comment by tebjan** [comment, marked as answer]
> * Is there anything to take in account for STRIDE in particular if I am going to re-use some existing C++  code . I am thinking Interop or Creating a wrapper like other type of apps in C# consuming C++ code.

Nothing special there, you can use any P/Invoke or native interop that C# has to offer, Stride is a normal .NET project.

> * I did not see anything about multi-player online games in Stride. So, do I have to build the server from scratch ? what would be the recommended way to do it ? I am looking for a simple server API to manage connections, and have a good set of primitives for messaging and event publishing. If there  is nothing, then I might look for message broker or an actor framework (my preferred).

There is nothing built-in, but Stride types are serializable and you are free to choose any networking solution from the whole .NET ecosystem. You can use game-specific frameworks like [lidgren](https://github.com/lidgren/lidgren-network-gen3) or whatever some internet research will give you. Here are a few more:

https://github.com/RevenantX/LiteNetLib
https://github.com/BeardedManStudios/ForgeNetworkingRemastered
https://github.com/tom-weiland/RiptideNetworking
https://github.com/Cysharp/MagicOnion
